File: src/main/java/dev/wiisportsresorts/duelsplugin/listeners/SumoLoseListener.java
package dev.wiisportsresorts.duelsplugin.listeners;

import dev.wiisportsresorts.duelsplugin.Duels;
import dev.wiisportsresorts.duelsplugin.util.McStyling;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

public class SumoLoseListener implements Listener {
  @EventHandler
  public void onMove(PlayerMoveEvent event) {
    Player player = event.getPlayer();
    if (Duels.sumo.current() == null) return;
    if (Duels.sumo.current().includesPlayer(player)) {
      if (player.getLocation().getBlock().isLiquid()) {
        Player winner = Duels.sumo.current().not(player);
        player.sendMessage(McStyling.format("$redYou lost the sumo duel against $gold%s$red!", winner.getName()));
        winner.sendMessage(McStyling.format("$greenYou won the sumo duel against $gold%s$green!", player.getName()));
        Duels.sumo.end(winner.getName(), player.getName());
      }
    }
  }
}

File: src/main/kotlin/dev/wiisportsresorts/duelsplugin/listeners/SumoLoseListener.kt
package dev.wiisportsresorts.duelsplugin.listeners

import dev.wiisportsresorts.duelsplugin.duels.SumoDuel
import dev.wiisportsresorts.duelsplugin.util.McStyling.gold
import dev.wiisportsresorts.duelsplugin.util.McStyling.green
import dev.wiisportsresorts.duelsplugin.util.McStyling.red
import org.bukkit.Sound
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class SumoLoseListener : Listener {
    @EventHandler
    fun onMove(event: PlayerMoveEvent) {
        if (SumoDuel.current == null) return
        val loser = event.player
        if (SumoDuel.current!!.includesPlayer(loser)) {
            if (loser.location.block.isLiquid) {
                val winner = SumoDuel.current!!.not(loser)

                SumoDuel.endWon(winner!!, loser)
            }
        }
    }
}